### DISCUSSION TITLE
virtualbox: cannot access ~/pCloudDrive

### DISCUSSION THREAD

**Discussion by pcolamar** [netblue30/firejail#5834] [Q&A] [answered]
Hi,
  What should I whitelist in an application  profile (e.g.: virtualbox)  to get the Vm to access ~/pCloudDrive ?
  
  Thanks

**Comment by ghost** [comment, marked as answer]
This usually is a two-step process. First step: protect ~/pCloudDrive from being accessed in general. You can do this by adding `blacklist ${HOME}/pCloudDrive` in a ~/.config/firejail/disable-common.local. The second step is to lift that blocked access in profiles you want to be able to see the path in a firejail sandbox. For non-whitelisting profiles adding `noblacklist ${HOME}/pCloudDrive` in a local override for the profile would suffice. For whitelisting profiles (like /etc/firejail/virtualbox.profile)  an _additional_ `whitelist ${HOME}/pCloudDrive` option is necessary.

After taking care of the above Virtualbox should be able to access your ~/pCloudDrive.

**Reply by pcolamar** [reply]
Close but no cigar @glitsj16 .
I might not have understood completely the whitelist/blacklist/noblacklist thing.
Sorry, 1st time I play with firejail  :-( 

the dir's I need to access are :

${HOME}/pCloudDrive/YPTO-Docs/
${HOME}/pCloudDrive/Ricerca-lavoro/

I have added the 2 files you suggested to create here below

Any additional hint ?

[disable-common.local.zip
[virtualbox.profile.zip](https://github.com/netblue30/firejail/files/11585302/virtualbox.profile.zip)
](https://github.com/netblue30/firejail/files/11585301/disable-common.local.zip)

Cheers
Palmer

**Comment by ghost** [comment]
Let's try again with the below files:

```sh
$ cat ~/.config/firejail/disable-common.local
# Firejail :: persistent local customizations for disable-common.inc

blacklist ${HOME}/pCloudDrive
```

```sh
$ cat ~/.config/firejail/virtualbox.local
# Firejail profile for virtualbox
# Persistent local customizations

noblacklist ${HOME}/pCloudDrive
whitelist ${HOME}/pCloudDrive
```

Please post the output of these commands:

```sh
$ firejail --ignore=quiet --name=vbox /usr/bin/virtualbox
```

After running the above, run this from another terminal (tab/window):

```sh
$ firejail --ls=vbox ~
```

**Reply by pcolamar** [reply]
Here you go.
In sequence log1 and log2

[log1.zip](https://github.com/netblue30/firejail/files/11585825/log1.zip)
I have noticed that 1n log1 there are a lot of errors though

[log2.zip](https://github.com/netblue30/firejail/files/11585826/log2.zip)

**Reply by ghost** [reply]
First thing I noticed in log1: `~/.config/firejail/virtualbox.local` isn't being included. Did you create that file with the aforementioned content? Looks like we need additional options to get around those many errors referring to /sys/module. Easiest way to test again is making sure you have this exact content in ~/.config/firejail/virtualbox.local:

```
# Firejail profile for virtualbox
# Persistent local customizations

noblacklist ${HOME}/pCloudDrive
whitelist ${HOME}/pCloudDrive

noblacklist /sys/module
whitelist /sys/module
read-only /sys/module

ignore private-etc
```

After adding that, re-run both commands and re-post logs please.

**Reply by pcolamar** [reply]
I might have misunderstood some of the earlier steps you suggested me.

Clean slate :
I have created the 2  .local files containing only what you wrote just above.
[disable-common.local.zip](https://github.com/netblue30/firejail/files/11593069/disable-common.local.zip)
[virtualbox.local.zip](https://github.com/netblue30/firejail/files/11593070/virtualbox.local.zip)

But with that I get this error 
[Log1-error.zip](https://github.com/netblue30/firejail/files/11593078/Log1-error.zip)

On the contrary , if I do not include the virtualbox.local you have attached above the application starts and the output of the 2 commands are:
[Log2.zip](https://github.com/netblue30/firejail/files/11593099/Log2.zip)
[Log3.zip](https://github.com/netblue30/firejail/files/11593100/Log3.zip)

no changes have been done to the default  /etc/firejail/virtualbox.profile (attached here below for your convenience)
[virtualbox.profile.zip](https://github.com/netblue30/firejail/files/11593849/virtualbox.profile.zip)

**Reply by pcolamar** [reply]
Post Scriptum
I have checked which one of the line in virtualbox.local is the culprit
`whitelist ${HOME}/pCloudDrive/`  it is.
if I comment it out( #, in front of it ) Virtualbox starts.

**Reply by pcolamar** [reply]
PPS.
Getting closer but still "no cigar"

Following some suggestion on other fora I got my virtualbox.local simplified like that

[virtualbox.local.zip](https://github.com/netblue30/firejail/files/11593865/virtualbox.local.zip)

No disable-common.local

With that I get : to see the directories ~/pCloudeDrive  and ~/pCloudeDrive/YPTO-Docs 
`palmer@AMD-Mint:~$ firejail --ls=vbox ~/pCloudDrive/
drwx------ 1000     1000              60 .
drwx------ 1000     1000             240 ..
drwx------ 1000     1000              40 YPTO-Docs
`
But when I go in YPTO-Docs , I do not see any file

`palmer@AMD-Mint:~$ firejail --ls=vbox ~/pCloudDrive/YPTO-Docs/
drwx------ 1000     1000              40 .
drwx------ 1000     1000              60 ..
`
Shall I open another Q&A with those details ?

**Reply by rusty-snake** [reply]
I don't really follow here because of all the unnecessary zip files. Never mind is pCloudDrive a normal folder or something special like a FUSE mount?

**Reply by ghost** [reply]
> Never mind is pCloudDrive a normal folder or something special like a FUSE mount?

@rusty-snake You might be on to something. If OP is indeed using [pCloud](https://www.pcloud.com/eu), according to its [AUR info](https://aur.archlinux.org/packages/pcloud-drive) that does seem to rely on FUSE.

@pcolamar See [this](https://firejail.wordpress.com/documentation-2/basic-usage/?like_comment=579#encfs) for details on how to deal with FUSE based filesystems in firejail.

**Reply by pcolamar** [reply]
![image](https://github.com/netblue30/firejail/assets/42656582/acd030b4-5a29-4e04-b659-616c7f67d454)
I suspect it is FUSE.

I do not know how to check yet (going to query Google) but the .fs suffix ( see screenshot)  let me think so.
I will read @glitsj16  doc to the link about FUSE

**Comment by pcolamar** [comment]
Small update,
 After an exchange with pCloud support it is clear to me that the solution is to implement what the  [link] (https://firejail.wordpress.com/documentation-2/basic-usage/?like_comment=579#encfs) suggested by @glitsj16 prpoposes.

But this can only be done using the cli version of pCloud . The appimage does not allow pass parameters when mounting FUSE.

 I will close the thread till I get the time to test.
 Thanks to all
